WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController+InformativeDialog.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+protocol GutenbergFlagsUserDefaultsProtocol {
+    func set(_ value: Bool, forKey defaultName: String)
+    func bool(forKey defaultName: String) -> Bool
+}
+
+extension UserDefaults: GutenbergFlagsUserDefaultsProtocol {
+
+}
+
+/// This extension handles Alert operations.
+extension GutenbergViewController {
+
+    enum Const {
+        enum Key {
+            static let informativeDialog = "Gutenberg.InformativeDialog"
+        }
+        enum Alert {
+            static let message = NSLocalizedString(
+                "The post was originally created in the Block Editor, so we've also enabled it on this Post. Switch back to Classic at any time by tapping ••• in the top bar.",
+                comment: "Popup content about why this post is being opened in Block Editor"
+            )
+            static let title = NSLocalizedString("Block Editor Enabled", comment: "Popup title about why this post is being opened in Block Editor")
+            static let okButtonTitle   = NSLocalizedString("OK", comment: "OK button to close the informative dialog on Gutenberg editor")
+        }
+    }
+
+    func showInformativeDialogIfNecessary() {
+        GutenbergViewController.showInformativeDialogIfNecessary(showing: post, on: self)
+    }
+
+    static func showInformativeDialogIfNecessary(
+        using userDefaults: GutenbergFlagsUserDefaultsProtocol = UserDefaults.standard,
+        showing post: AbstractPost,
+        on viewController: UIViewControllerTransitioningDelegate & UIViewController,
+        animated: Bool = true
+    ) {
+        guard !userDefaults.bool(forKey: Const.Key.informativeDialog),
+            post.containsGutenbergBlocks() else {
```

Comment:
hey @etoledom I've put an extra check `post.containsGutenbergBlocks()` check here, although this might be a bit unnecessary since we only open Gutenberg editor for blocks that contain blocks, but couldn't quite foresee the future, wdyt?

Reply:
Yes, it seems a bit redundant but it wouldn't hurt to be extra careful. I'd say let's keep it 👍